include/ccf/crypto/cose_verifier.h

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ccf/crypto/key_pair.h"
+#include "ccf/crypto/pem.h"
+#include "ccf/crypto/public_key.h"
+
+#include <chrono>
+#include <t_cose/q_useful_buf.h>
+
+namespace crypto
+{
+  class COSEVerifier
+  {
+  public:
+    virtual bool verify(
+      const q_useful_buf_c& buf, q_useful_buf_c& authned_content) const = 0;
```

Comment:
Can we use `std::span` or similar here, rather than the custom type? Or have we already got the custom type from one of their APIs, that we're trying to pass to another?

Reply:
Either way it goes to/comes from their API, but we could wrap everything in one level of `std::span`.